opensrp-app/src/main/java/org/smartregister/commonregistry/CommonRepository.java

```diff
@@ -302,7 +341,7 @@ private ContentValues createValuesFor(CommonPersonObject common) {
             int columncount = cursor.getColumnCount();
             HashMap<String, String> columns = new HashMap<String, String>();
             for (int i = initialColumnCount; i < columncount; i++) {
-                columns.put(additionalcolumns[i - initialColumnCount], cursor.getString(i));
+                columns.put(additionalcolumns[i - initialColumnCount].getName(), cursor.getString(i));
```

Comment:
I think this might not be fetchable as String in cases where the column is not of type `VARCHAR`

This also applies for other cases such as https://github.com/OpenSRP/opensrp-client-core/blob/b78069db324c1c07eb42a1826197c8882b271692/opensrp-app/src/main/java/org/smartregister/commonregistry/CommonRepository.java#L501

Reply:
This is just being used to Return the Name. Not the dataType

Reply:
Most values are able to be returned as String (Integer, Real, Text) all respect the `toString()`, apart from Blob datatype. We can return the value as as Object but that would break implementation that assume the common person object stores Strings. Also I'm not sure if we have any plans to store Blobs in the database